banyan/tests/common.rs

```diff
@@ -93,6 +126,57 @@ pub fn txn(store: MemStore<Sha256Digest>, cache_cap: usize) -> Txn {
     Txn::new(Forest::new(store.clone(), branch_cache), store)
 }
 
+#[derive(Debug, Clone)]
+pub struct TestFilter {
+    offset_range: Range<u64>,
+    keys: RangeSet<u64>,
+}
+
+impl TestFilter {
+    pub fn offset_range(offset_range: Range<u64>) -> Self {
+        Self {
+            offset_range,
+            keys: RangeSet::all(),
+        }
+    }
+
+    pub fn keys(values: impl IntoIterator<Item = u64>) -> Self {
+        let mut keys = RangeSet::empty();
+        for value in values {
+            keys |= RangeSet::from(value..value + 1);
+        }
+        Self {
+            offset_range: 0..u64::max_value(),
+            keys,
+        }
+    }
+
+    pub fn query(&self) -> impl Query<TT> + Clone {
+        AndQuery(
+            OffsetRangeQuery::from(self.offset_range.clone()),
+            KeyQuery(self.keys.clone()),
+        )
+    }
+
+    pub fn contains(&self, triple: &(u64, Key, u64)) -> bool {
```

Comment:
why not just the offset+key tuple?

Reply:
Because the triple is what comes out of the usual tree methods. But I guess you would never need the value, since queries do not have access to values.